src/test/java/hillel/spring/DoctorRestAPI/DoctorRestAPIControllerTest.java

```diff
@@ -0,0 +1,141 @@
+package hillel.spring.DoctorRestAPI;
+
+import org.assertj.core.api.Assertions;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.util.ResourceUtils;
+
+import java.io.File;
+import java.nio.file.Files;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class DoctorRestAPIControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    DoctorRestAPIRepo doctorRestAPIRepo;
+
+    @After
+    public void cleanUp(){
+        doctorRestAPIRepo.deleteAll();
+    }
+
+    @Test
+    public void shouldFindAllDoctors() throws Exception {
+        doctorRestAPIRepo.init();
+
+        mockMvc.perform(get("/doctors"))
+               .andExpect(status().isOk())
+               .andExpect(content().json(fromResource("DoctorRestAPI/all-doctors.json")));
+    }
+
+    @Test
+    public void shouldFindDoctorById() throws Exception {
+        doctorRestAPIRepo.createDoctor(new Doctor(1,"Aaa","surgeon"));
+        doctorRestAPIRepo.createDoctor(new Doctor(3,"ccc","NoSurgeon"));
+
+        mockMvc.perform(get("/doctors/{id}",1))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\n" +
+                        "    \"id\": 1,\n" +
+                        "    \"name\": \"Aaa\",\n" +
+                        "    \"specialization\": \"surgeon\"\n" +
+                        "  }"));
```

Comment:
I would recommend moving this JSON to .json file since escaped JSON is not readable at all